packages/mdc-select/index.ts

```diff
@@ -0,0 +1,596 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+import {CustomEventListener, MDCComponent} from '@material/base/index';
+import {EventType, SpecificEventListener} from '@material/base/index';
+import {MDCFloatingLabel} from '@material/floating-label/index';
+import {MDCLineRipple} from '@material/line-ripple/index';
+import * as menuSurfaceConstants from '@material/menu-surface/constants';
+import * as menuConstants from '@material/menu/constants';
+import {MDCMenu, MenuItemEvent} from '@material/menu/index';
+import {MDCNotchedOutline} from '@material/notched-outline/index';
+import {MDCRipple, MDCRippleFoundation, RippleCapableSurface} from '@material/ripple/index';
+import {cssClasses, strings} from './constants';
+import {MDCSelectFoundation} from './foundation';
+import {MDCSelectHelperText} from './helper-text';
+import {MDCSelectIcon} from './icon';
+import {
+  FoundationMapType,
+  HelperTextFactory, IconFactory,
+  LabelFactory,
+  LineRippleFactory, MenuFactory,
+  OutlineFactory, SelectEventDetail,
+} from './types';
+
+const VALIDATION_ATTR_WHITELIST = ['required', 'aria-required'];
+const MOUSEDOWN_TOUCHSTART_EVENTS: EventType[] = ['mousedown', 'touchstart'];
+
+class MDCSelect extends MDCComponent<MDCSelectFoundation> implements RippleCapableSurface {
+  static attachTo(root: Element): MDCSelect {
+    return new MDCSelect(root);
+  }
+
+  // Public visibility for this property is required by RippleCapableSurface.
+  root_!: HTMLElement; // assigned in MDCComponent constructor
+
+  private menu_!: MDCMenu | null; // assigned in enhancedSelectSetup_()
+  private menuOpened_ = false;
+
+  // Exactly one of these fields must be non-null.
+  private nativeControl_!: HTMLSelectElement | null;
+  private selectedText_!: HTMLElement | null;
+
+  private hiddenInput_!: HTMLInputElement | null;
+  private leadingIcon_!: MDCSelectIcon | null;
+  private helperText_!: MDCSelectHelperText | null;
+  private menuElement_!: Element | null;
+  private ripple!: MDCRipple | null;
+  private lineRipple_!: MDCLineRipple | null;
+  private label_!: MDCFloatingLabel | null;
+  private outline_!: MDCNotchedOutline | null;
+  private handleChange_!: SpecificEventListener<'change'>; // assigned in initialize()
+  private handleFocus_!: SpecificEventListener<'focus'>; // assigned in initialize()
+  private handleBlur_!: SpecificEventListener<'blur'>; // assigned in initialize()
+  private handleClick_!: SpecificEventListener<'click'>; // assigned in initialize()
+  private handleKeydown_!: SpecificEventListener<'keydown'>; // assigned in initialize()
+  private handleMenuOpened_!: EventListener; // assigned in initialize()
+  private handleMenuClosed_!: EventListener; // assigned in initialize()
+  private handleMenuSelected_!: CustomEventListener<MenuItemEvent>; // assigned in initialize()
+  private validationObserver_!: MutationObserver; // assigned in initialize()
+
+  get value(): string {
```

Comment:
I'm a little confused why the getters/setters are before initialize/destroy/etc. this time? I guess this might be consistent with the original but if git is going to consider it a rewrite anyway, we might as well make this more consistent with the others.

Reply:
Done